query/src/main/java/org/infinispan/query/impl/QueryDefinition.java

```diff
@@ -207,7 +226,8 @@ public void writeObject(ObjectOutput output, QueryDefinition queryDefinition) th
             output.writeObject(queryDefinition.queryEngineProvider);
          } else {
             output.writeBoolean(false);
-            output.writeObject(queryDefinition.hsQuery);
+            output.writeObject(queryDefinition.hsQuery.getLuceneQuery());
```

Comment:
What about the other serializable parts of ```hsQuery```?

Reply:
Do you have any fields in mind? Looking at the non-transient fields in LuceneHSQuery impl I don't think we need to serialiaze any of the fields other than the LuceneQuery in order to recreate the state.

Reply:
That are other types of hsQuery unfortunately. I am a bit puzzled no test failed by ignoring the serialization of the other stuff, let me check...

Reply:
> That are other types of hsQuery unfortunately.

True, that's actually partly why I thought that recreating the HSQuery via the LuceneQuery and `SearchIntegrator.createHSQuery` was sufficient, as the `SearchIntegrator` only provides `createHSQuery(Query, Class<?>...)` and ``createHSQuery(Query, IndexedTypeMap<CustomTypeMetadata>)`.

Reply:
```Internal and experimental! Creates a {@link CacheQuery}, filtered according to the given {@link HSQuery}.```
```SearchManagerImpl#getQuery(HSQuery hSearchQuery, IndexedQueryMode queryMode)```

That's where it is used. It's an internal stuff, not part of the SearchManager interface

Reply:
So if someone is using this, it has already been warned :)  WDYT @anistor ?